### PR TITLE
Add TableHandler::declare_column().

### LINKED ISSUE #4227: TableHandler improvements
https://github.com/geodynamics/aspect/pull/1429 shows a number of oddities in the `TableHandler` class that we've known of for a while already but that are still worth addressing:

* Provide a "canonical" way to start a new row in the table, rather than have some magic in `TableHandler::add_value` that does this.
* Provide a way to "declare" a column, rather than creating it the first time it is used. This would help with  being able to describe a particular order for columns we may want to have when creating output, and keeping this order even if the code that fills the table moves around within a piece of software.

### PR DESCRIPTION
**Pull request by bangerth**
Closes #4227.

### CODE REVIEW

File: source/base/table_handler.cc [hunk 1]
```diff
@@ -189,6 +189,18 @@ TableHandler::TableHandler()
...
+  // see if the column already exists; insert it if not
+  Assert (columns.find(key) == columns.end(),
+          ExcMessage ("You are trying to declare a column with key <" + key +
+                      "> but such a column already exists."));
+
+  columns.insert(std::make_pair(key, Column(key)));
+  column_order.push_back(key);
+}
+
+
```
Comment (masterleinad) [inline source/base/table_handler.cc]: Can we have three blank lines? :wink:

File: tests/base/table_handler_15.cc [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+#include <iomanip>
+#include <fstream>
+#include <string>
+
+
+int main ()
+{
+  std::ofstream logfile("output");
+  deallog.attach(logfile);
+  deallog.threshold_double(1.e-10);
```
Comment (masterleinad) [inline tests/base/table_handler_15.cc]: We can just use `initlog()`.

**Comment by bangerth** [comment]
I fixed the two issues.

**Comment by kronbichler** [comment]
The tester (or its submission) seems to have stopped in the middle, but everything appears sane, so I'm merging now.

**Pull request merged**